src/NHibernate.Test/NHSpecificTest/GH2608/Fixture.cs

```diff
@@ -0,0 +1,33 @@
+﻿using NUnit.Framework;
+
+namespace NHibernate.Test.NHSpecificTest.GH2608
+{
+	[TestFixture]
+	public class Fixture : BugTestCase
+	{
+		protected override void OnTearDown()
+		{
+			using (var session = OpenSession())
+			using (var transaction = session.BeginTransaction())
+			{
+				session.CreateQuery("delete from PersonalDetails").ExecuteUpdate();
+				session.CreateQuery("delete from System.Object").ExecuteUpdate();
+
+				transaction.Commit();
+			}
+		}
+
+		[Test]
+		public void MergeBidiPrimaryKeyOneToOne()
+		{
+			using (var s = OpenSession())
+			using (var tx = s.BeginTransaction())
+			{
+				var p = new Person { Name = "steve" };
+				p.Details = new PersonalDetails { SomePersonalDetail = "I have big feet", Person = p };
+				s.Merge(p);
```

Comment:
It looks like the same exception should be thrown on `Persist` too. Am I right?

Reply:
No, persist does not fail, it least with my tests on a modified MergeFixture.

Reply:
Strange... I thought it's directly related to #1754 in which case Persist should fail the same (at least for this test case). I added test for Persist - let's see...

Reply:
>  it least with my tests on a modified MergeFixture.

But in current state it also doesn't fail on Merge.... The only failed tests are newly added GH2608

Reply:
The `MergeFixture` one fails if we change its id generator from `increment` to `identity`. But as this would mean changing from a app-side generated id to a db-side one, I have not done that change, it would change too much the test.

I have re-checked these GH2608 tests on 5.2.x just in case, both succeed on 5.2.x.